### PR TITLE
Integration tests for ingress and Modsec to include dedicated ingress controller

### PR DESCRIPTION
**Pull request by poornima-krishnasamy**
related to: https://github.com/ministryofjustice/cloud-platform/issues/2246

### CODE REVIEW

**Review by digitalronin — DISMISSED** [review]

File: smoke-tests/spec/modsec_default_spec.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
+require "spec_helper"
+
+# The spec in this file is for testing ModSecurity with default "nginx"
+# ingress controller by using the ingress_class = "nginx" in their ingress annotation
+
+describe "Testing modsec" do
```
Comment (jasonBirchall) [inline smoke-tests/spec/modsec_default_spec.rb:6]: Starting to think we should change the description to Testing modsec on ingress class: "nginx"

File: smoke-tests/spec/modsec_default_spec.rb [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    )
+    wait_for(namespace, "ingress", ingress_name)
+    sleep 120 # We need to wait for a while *after* the ingress is created before we try to test it, or we get failures.
+  end
+
+  after(:all) do
+    delete_namespace(namespace)
+  end
+
+  context "when modsec deployed with nginx ingress class" do # this is the default behaviour
```
Comment (jasonBirchall) [inline smoke-tests/spec/modsec_default_spec.rb:34]: The above change would mean the context would be "when modsec is deployed"

File: smoke-tests/spec/modsec_default_spec.rb [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    context "when the url is malicious" do
+      let(:url) { bad_url }
+
+      specify "request is blocked" do
+        expect { URI.open(url) }.to raise_error(OpenURI::HTTPError, "403 Forbidden")
+      end
+    end
+  end
+
+  context "when modsec disabled  with nginx ingress class" do
```
Comment (jasonBirchall) [inline smoke-tests/spec/modsec_default_spec.rb:58]: Same as above

File: smoke-tests/spec/modsec_spec.rb [hunk 4]
```diff
@@ -1,9 +1,15 @@
 require "spec_helper"
 
-xdescribe "Testing modsec" do
+# The spec in this file is for testing ModSecurity with dedicated ingress controller.
+# Integration tests has the dedicated ingress controller with ingress class - integration-test
+# The below spec uses the ingress_class = "integration-test" in its ingress annotation
+
+describe "Testing modsec" do
```
Comment (jasonBirchall) [inline smoke-tests/spec/modsec_spec.rb:7]: This would be "Testing modsec on ingress class: integration-test"

**Review by jasonBirchall — APPROVED** [review]

**Pull request merged**